tsfc/kernel_interface/firedrake_loopy.py

```diff
@@ -173,7 +175,8 @@ def construct_kernel(self, return_arg, impero_c, index_names):
         loopy_kernel = generate_loopy(impero_c, args, self.scalar_type,
                                       "expression_kernel", index_names)
         return ExpressionKernel(loopy_kernel, self.oriented, self.cell_sizes,
-                                self.coefficients, self.tabulations)
+                                self.coefficients, first_coefficient_fake_coords,
```

Comment:
Another thing I think we should think about next time is how to make `ExpressionKernelBuilder` more symmetric to `KernelBuilder`. I think instead of returning `coefficients` here, we should just call `extract_coefficients(expr)` again in Firedrake (this compares to `form.coefficients()` called in `assemble.py`)  and split `Function`s if necessary when setting up `ParLoop` arguments (again, as is done in `assemble.py`).